docs/configuration/prometheus-config.md

```diff
@@ -61,15 +61,20 @@ agents distribute discovery and scrape load between nodes.
 # cannot be used.
 [enabled: <boolean> | default = false]
 
-# How often should the agent manually reshard. Useful for if KV change
+# Note these next 3 configuration options are confusing. Due to backwards compatibility the naming
+# is less than ideal.
+
+# How often should the agent manually refresh the configuration. Useful for if KV change
 # events are not sent by an agent.
 [reshard_interval: <duration> | default = "1m"]
 
-# The timeout for a reshard. Applies to a cluster-wide reshard (done when
-# joining or leaving the cluster) and local reshards (done every
-# reshard_interval). A timeout of 0 indicates no timeout.
+# The timeout for configuration refreshes. This can occur on cluster events or 
+# on the reshard interval. A timeout of 0 indicates no timeout.
 [reshard_timeout: <duration> | default = "30s"]
 
+# The timeout for a cluster reshard events. A timeout of 0 indicates no timeout.
+[refresh_timeout: <duration> | default = "30s"]
```

Comment:
nit: What about `cluster_reshard_event_timeout`? Your comment seems to nail the description.